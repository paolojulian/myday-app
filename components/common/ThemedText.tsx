import { useMemo } from 'react';
import { StyleSheet, Text, TextProps } from 'react-native';

type ThemedText = {
  variant?: 'heading' | 'body' | 'body-lg';
} & TextProps;

export default function ThemedText({ variant = 'body', ...props }: ThemedText) {
  const variantStyle = useMemo(() => {
    return styles[variant] || styles.body;
  }, [variant]);

  return <Text {...props} style={[variantStyle, styles.baseStyle]}></Text>;
}

const styles = StyleSheet.create({
  baseStyle: {
    fontFamily: 'Inter',
  },
  heading: {
    fontSize: 48,
  },
  body: {
    fontSize: 16,
  },
  ['body-lg']: {
    fontSize: 24,
    fontWeight: '500',
  },
});
