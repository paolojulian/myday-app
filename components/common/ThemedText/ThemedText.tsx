import { useMemo } from 'react';
import { StyleSheet, Text, TextProps } from 'react-native';

type ThemedTextProps = {
  variant?: 'heading' | 'body' | 'body1' | 'body2';
} & TextProps;

export default function ThemedText({ variant = 'body', style, ...props }: ThemedTextProps) {
  const variantStyle = useMemo(() => {
    return themedTextStyles[variant] || themedTextStyles.body;
  }, [variant]);

  return <Text {...props} style={[variantStyle, themedTextStyles.baseStyle, style]}></Text>;
}

export const themedTextStyles = StyleSheet.create({
  baseStyle: {
    fontWeight: 400,
  },
  caps: {
    fontSize: 64,
    fontWeight: 800,
    fontFamily: 'Barlow_CondensedBold',
  },
  heading: {
    fontSize: 32,
    fontWeight: 600,
    fontFamily: 'Open_SansSemiBold',
  },
  body: {
    fontSize: 14,
    fontFamily: 'Open_SansRegular',
  },
  body1: {
    fontSize: 16,
    fontFamily: 'Open_SansRegular',
  },
  body2: {
    fontSize: 16,
    fontWeight: 500,
    fontFamily: 'Open_SansMedium',
  },
});
