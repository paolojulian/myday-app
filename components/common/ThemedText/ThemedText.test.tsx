import ThemedText from '@/components/common/ThemedText/ThemedText';
import { render } from '@testing-library/react-native';

describe('TESTING ThemedText Component', () => {
  describe('WHEN the component is rendered', () => {
    it('should match the snapshot', () => {
      const container = render(<ThemedText>Test</ThemedText>);
      expect(container).toMatchSnapshot();
    });
    it('should render without crashing', () => {
      render(<ThemedText>Test</ThemedText>);
      expect(ThemedText).toBeTruthy();
    });
  });

  describe('WHEN the variant is set to heading', () => {
    it('should match the snapshot', () => {
      const container = render(<ThemedText variant="heading">Test</ThemedText>);
      expect(container).toMatchSnapshot();
    });
    it('should render without crashing', () => {
      render(<ThemedText>Test</ThemedText>);
      expect(ThemedText).toBeTruthy();
    });
  });
});
