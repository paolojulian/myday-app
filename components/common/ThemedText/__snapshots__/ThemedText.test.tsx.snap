// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING ThemedText Component WHEN the component is rendered should match the snapshot 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "PoppinsRegular",
        "fontSize": 14,
        "lineHeight": 20,
      },
      {
        "fontWeight": 400,
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`TESTING ThemedText Component WHEN the variant is set to heading should match the snapshot 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "PoppinsSemiBold",
        "fontSize": 24,
        "fontWeight": 600,
        "lineHeight": 32,
      },
      {
        "fontWeight": 400,
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;
